cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactMeta.java

```diff
@@ -18,24 +18,39 @@
 
 import co.cask.cdap.api.templates.plugins.PluginClass;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
 
+import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
+import java.util.Set;
 
 /**
- * Metadata about an artifact, such as what plugins are contained in the artifact.
+ * Metadata about an artifact, such as what plugins are contained in the artifact, and what other artifacts can use
+ * the plugins in this artifact.
```

Comment:
hm thats a good question.  The current code would not let you (unless its a snapshot), but it may make sense. I think to start out with we will not allow it, but could relax that later on.